core/primitives/src/block.rs

```diff
@@ -86,7 +86,8 @@ impl Block {
         header: BlockHeader,
         body: BlockBody,
     ) -> Block {
-        if !ProtocolFeature::BlockHeaderV4.enabled(this_epoch_protocol_version) {
+        #[allow(deprecated)]
+        if !ProtocolFeature::_DeprecatedBlockHeaderV4.enabled(this_epoch_protocol_version) {
```

Comment:
While working with deprecated protocol features, assume protocol_version is always greater than the feature protocol version, i.e. the feature is always enabled